src/hangar/cli/cli.py

```diff
@@ -535,136 +536,172 @@ def server(overwrite, ip, port, timeout):
-@click.option('-s', '--sample', default=False, help='Sample name to export')
-@click.option('-f', '--format', 'format_', required=False, help='File format used for exporting.')
+@click.argument('startpoint', nargs=1, default=None, required=False)
+@click.option('-o', '--out', 'outdir', required=False, default=os.getcwd(), help="Directory to export data")
+@click.option('-s', '--sample', default=None, help='Sample name to export. You can optionally specify the type'
+                                                   ' of sample name after the colon (ex. `54:str` or `54:int`)')
+@click.option('-f', '--format', 'format_', required=False, help='File format of output file')
 @click.option('--plugin', required=False, help='override auto-inferred plugin')
 @pass_repo
-def export_data(repo: Repository, startpoint, arrayset, out, sample, format_, plugin):
-    """export ARRAYSET sample data as it existed a STARTPOINT to some format and path.
+@click.pass_context
+def export_data(ctx, repo: Repository, arrayset, outdir, startpoint, sample, format_, plugin):
+    """Export ARRAYSET sample data as it existed a STARTPOINT to some format and path.
+
+    Specifying which sample to be exported is possible by using the switch
+    `--sample` (without this, all the samples in the given arrayset will be
+    exported). Since hangar supports both int and str datatype for the sample
+    name, specifying that while mentioning the sample name might be necessary
+    at times. It is possible to do that by separating the name and type by a
+    colon.
+
+    Example:
+
+       1. if the sample name is string of numeric 10 - ``10:str`` or ``10``
+
+       2. if the sample name is ``sample1`` - ``sample1:str1`` or ``sample1``
+
+       3. if the sample name is an int, let say 10 - ``10:int``
     """
+    # TODO: ignore warning through env variable
     from hangar.records.commiting import expand_short_commit_digest
-    from hangar.records.heads import get_branch_head_commit
-    from hangar.cli.io import imsave
+    from hangar.records.heads import get_branch_head_commit, get_staging_branch_head
+    from hangar import external
+
+    kwargs = parse_custom_arguments(ctx.args)
 
     if startpoint in repo.list_branches():
         base_commit = get_branch_head_commit(repo._env.branchenv, startpoint)
-    else:
+    elif startpoint:
         base_commit = expand_short_commit_digest(repo._env.refenv, startpoint)
+    else:
+        branch_name = get_staging_branch_head(repo._env.branchenv)
+        base_commit = get_branch_head_commit(repo._env.branchenv, branch_name)
+    co = repo.checkout(commit=base_commit)
 
     try:
-        co = repo.checkout(write=False, commit=base_commit)
-        arrayset = co.arraysets[arrayset]
+        aset = co.arraysets.get(arrayset)
         if sample:
+            sample, stype = sample.split(':') if ':' in sample else (sample, '')
+            sample = int(sample) if stype == 'int' else str(sample)
```

Comment:
Instead of doing this same parsing logic in every one of the plugin methods, you should be using a custom parameter type. That way you could just say 

```python
@click.option('-s', '--sample',
               type=utils.StrOrIntType, 
               default=None, 
                help='Sample name to export. You can optionally specify the type'
                      ' of sample name after the colon (ex. `54:str` or `54:int`)')
```

and each element in the sample list would be automatically parsed and converted by that function before it even hits our function here. 

Please read up on it: https://click.palletsprojects.com/en/7.x/parameters/#implementing-custom-types

Reply:
Nice stuff!